src/symbols.cpp

```diff
@@ -91,11 +100,61 @@ void Symbols::parseKernelSymbols(CodeCache* cc) {
             }
         }
     }
+}
+
+void Symbols::parseLibrarySymbols(CodeCache* cc, const char* lib_name, const char* base) {
+    // Parse symbols from the original .so
+    // If there is no .strtab section, try loading symbols from an external debug library
```

Comment:
I think it is actually preferable to load symbols from external debuginfo first, if it is present, than from the actual binary. The reason is that the binary might have stripped symbols, or partial symbols, while the external debuginfo is very likely to have more complete symbol information.